codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/PresignURLAutoFill.java

```diff
@@ -71,6 +66,122 @@
+            GoWriter writer,
+            SymbolProvider symbolprovider,
+            OperationShape operation,
+            StructureShape input,
+            MemberShape member
+    ) {
+        Symbol operationSymbol = symbolprovider.toSymbol(operation);
+        Symbol inputSymbol = symbolprovider.toSymbol(input);
+        String memberName = symbolprovider.toMemberName(member);
+
+        writer.openBlock("func $L(params interface{}, value string) error {", "}",
+                setterFuncName(operationSymbol.getName(), memberName),
+                () -> {
+                    writer.addUseImports(SmithyGoDependency.FMT);
+                    writer.write("input, ok := params.($P)", inputSymbol);
+                    writer.openBlock("if !ok {", "}", () -> {
+                        writer.write("return fmt.Errorf(\"expect $P type, got %T\", params)", inputSymbol);
+                    });
+                    writer.write("input.$L = &value", memberName);
+                    writer.write("return nil");
+                });
+    }
+
+    private static void writeMemberGetter(
+            GoWriter writer,
+            SymbolProvider symbolprovider,
+            OperationShape operation,
+            StructureShape input,
+            MemberShape member
+    ) {
+        Symbol operationSymbol = symbolprovider.toSymbol(operation);
+        Symbol inputSymbol = symbolprovider.toSymbol(input);
+        String memberName = symbolprovider.toMemberName(member);
+
+        writer.openBlock("func $L(params interface{}) (string, bool, error) {", "}",
+                getterFuncName(operationSymbol.getName(), memberName),
+                () -> {
+                    writer.addUseImports(SmithyGoDependency.FMT);
+                    writer.write("input, ok := params.($P)", inputSymbol);
+                    writer.openBlock("if !ok {", "}", () -> {
+                        writer.write("return ``, false, fmt.Errorf(\"expect $P type, got %T\", params)", inputSymbol);
+                    });
+                    writer.openBlock("if input.$L == nil || len(*input.$L) == 0 {", "}", memberName, memberName,
+                            () -> writer.write("return ``, false, nil")
+                    );
+                    writer.write("return *input.$L, true, nil", memberName);
+                });
+    }
+
+    private static void writePresignOperationAccessor(
+            GoWriter writer,
+            SymbolProvider symbolprovider,
+            OperationShape operation,
+            StructureShape input
+    ) {
+        Symbol operationSymbol = symbolprovider.toSymbol(operation);
+        Symbol inputSymbol = symbolprovider.toSymbol(input);
+
+        Symbol removeMiddleware = SymbolUtils.createValueSymbolBuilder("RemoveMiddleware",
+                AwsCustomGoDependency.PRESIGNEDURL_CUSTOMIZATION).build();
+
+        writer.openBlock(
+                "func $L(ctx context.Context, client interface{}, region string, params interface{}) "
```

Comment:
Does the `client` need to be an empty interface? I'd think this would be specific to the `*PresignClient`.